packages/core/misc-widgets/src/components/github-star/GithubStar.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <div
+    v-if="url"
+    class="kong-ui-public-misc-widgets-github-star"
+    data-testid="github-star"
+  >
+    <a
+      aria-label="i18n.t('githubStar.ariaLabel')"
+      class="github-button"
+      data-color-scheme="no-preference: light; light: light; dark: light;"
+      data-show-count="true"
+      :href="url"
+      target="_blank"
+    >{{ i18n.t('githubStar.title') }}</a>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onMounted } from 'vue'
+import composables from '../../composables'
+
+defineProps({
+  url: {
+    type: String,
+    required: true,
+  },
+})
+
+const { i18n } = composables.useI18n()
+
+onMounted(async () => {
+  const githubStarScript = document.createElement('script')
+  githubStarScript.setAttribute('src', 'https://buttons.github.io/buttons.js')
+  document.head.appendChild(githubStarScript)
+})
```

Comment:
I think you could do something like this, and then add the `scriptLoaded` variable in your `v-if` check in the template (you'll also need to import `ref` from `vue`)

```suggestion
const scriptLoaded = ref<boolean>(false)

onMounted(async () => {
  const githubStarScript = document.createElement('script')
  githubStarScript.addEventListener('load', (event) => {
    scriptLoaded.value = true
  })
  githubStarScript.setAttribute('src', 'https://buttons.github.io/buttons.js')
  document.head.appendChild(githubStarScript)
})
```


script.addEventListener("load", function(event) {
    console.log("script loaded :)");
    onjqloaded(); // in fact, yourstuffscript() function
})

Reply:
True, tried out your solution, and it renders the GHStar only if the script is loaded.

<img width="436" alt="Screenshot 2023-02-15 at 12 26 40 PM" src="https://user-images.githubusercontent.com/2568272/219146881-5eed1acf-e229-4016-a63c-14ff837a85d6.png">